### ISSUE TITLE
Save File mode as instance variable

### ISSUE THREAD

**Issue by watzon** [crystal-lang/crystal#8922]
In some cases you may want to know what mode a file was opened in, but currently that doesn't seem to be possible since the mode is just passed to the FileDescriptor and forgotten about. It would be nice if we could get a `#mode` getter with a corresponding instance variable.

**Comment by RX14** [comment]
This is possible without storing them in an instance variable using `fcntl(F_GETFL)`

**Comment by watzon** [comment]
Good idea. It would just be nice to have it. A to_unsafe or similar method that returns a C compatible FILE would be nice too. I do a lot of C lib wrapping and this isn't the first time I've had to implement it myself.

**Comment by RX14** [comment]
See #3151 for `FILE*`.

**Comment by watzon** [comment]
Yeah I saw that issue. Not sure why we don't want it in LibC at least though. It would be nice to have LibC be as complete as possible, no?

**Comment by RX14** [comment]
No, we'll probably move `LibC` to `Crystal::LibC` eventually to make it clear it's not public. It's there for the stdlib only really, i think it's a better pattern to bind what your shard uses from libc in your own shard.
